nautobot/extras/tests/test_plugins.py

```diff
@@ -439,6 +440,37 @@ def test_api_urls(self):
         detail_url = reverse("plugins-api:example_plugin-api:examplemodel-detail", kwargs={"pk": instance.pk})
         self.assertEqual(detail_url, self._get_detail_url(instance))
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+    @mock.patch("nautobot.extras.api.serializers.reverse")
+    def test_get_notes_url_on_object_raise_no_reverse_match(self, mock_reverse):
+        # This test is to check the error handling when the user implemented a model without NotesMixin
+        # But include the NotesSerializerMixin in the Model APIViewSet with NautobotModelViewSet
+
+        # Raise NoReverseMatch in get_notes_url()
+        mock_reverse.side_effect = mock.Mock(side_effect=NoReverseMatch())
+        instance = self._get_queryset().first()
+        obj_perm = ObjectPermission(
+            name="Test permission",
+            constraints={"pk": instance.pk},
+            actions=["view"],
+        )
+        obj_perm.save()
+        obj_perm.users.add(self.user)
+        obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))
+        url = self._get_detail_url(instance)
+
+        # Check if log messages and API data is properly rendered
+        with self.assertLogs(logger="nautobot.extras.api", level="WARNING") as cm:
+            response = self.client.get(url, **self.header)
+            self.assertHttpStatus(response, 200)
+            self.assertIn('"notes_url":null', response.content.decode(response.charset))
```

Comment:
Having a hard time following this one. What's it looking for? Can we provide a more verbose match string?

Reply:
This was testing if `return None` is rendered correctly in the browsable API page as `"notes_url": null`. But I dont think its needed since we already do `self.assertIsNone(response.data["notes_url"])`
<img width="938" alt="Screen Shot 2022-11-08 at 10 21 14 PM" src="https://user-images.githubusercontent.com/46973263/201218033-a9ea4140-71aa-4eb4-b470-0fdb6d220dae.png">